### DISCUSSION TITLE
imported OBS scenes order / keyboard shortcuts

### DISCUSSION THREAD

**Discussion by riemsesy** [sircharlo/meeting-media-manager#293] [Ideas] [answered]
Hello Charles, is it possible to import the OBS scenes in JWMMF in the same order as they are configured in OBS?
As an example, we have the  WT reader and the WT overseer camera scenes as two different scenes as they are in two different locations on the stage.
In OBS these scenes are next to each other so we can toggle them with the arrow keys, as keyboard shortcuts. It makes it quicker to switch.
Because JWMMF orders the scenes alphabetically, they are not next to each other.

A solution can also be if all scenes have a keyboard shortcut like ALT+1, ALT+2 etc.. or any other key 

and / or I am not totally happy with the dropdown box for the camera scenes, but I don't know if there is place for a fixed list to select a scene, and because you never know how many scenes someone has configured.

**Comment by timjobutler** [comment]
Out of curiosity, how many scenes have been configured? According to the _Audio/​Video Instructions for Hybrid Meetings_ (S-261 dated 3/22) paragraph 8, it states:

> A single camera showing a wide shot of the stage is sufficient. ... While the exact position of the camera may vary according to local circumstances and equipment, endeavor to obtain a shot similar to that depicted below.

![Screenshot_20220508-101827.jpg](https://user-images.githubusercontent.com/37101212/167276443-f8a71271-b781-4bd9-8d3a-fb5eb12039fb.jpg)

Based on this direction, only one camera scene of the platform would be required in OBS.

**Comment by riemsesy** [comment]
Our circumstances differ. We have a very high ceiling and as per guidelines we have a PTZ camera installed. When we try to bring the complete stage in picture we will se heads of brs sisters or visitors.
So we zoom in on the speaker like in s178 (which was for camera setup at home)
[Image.jpeg]
We use this for the speaker and WT reader and for those giving presentations.
Our elder committee gave us green light for this setup.

Groeten, Richard
________________________________
Van: Tim Butler ***@***.***>
Verzonden: Sunday, May 8, 2022 2:26:13 AM
Aan: sircharlo/jw-meeting-media-fetcher ***@***.***>
CC: riemsesy ***@***.***>; Author ***@***.***>
Onderwerp: Re: [sircharlo/jw-meeting-media-fetcher] imported OBS scenes order / keyboard shortcuts (Discussion #293)


Out of curiosity, how many scenes have been configured? According to the Audio/​Video Instructions for Hybrid Meetings (S-261 dated 3/22) paragraph 8, it states:

A single camera showing a wide shot of the stage is sufficient. ... While the exact position of the camera may vary according to local circumstances and equipment, endeavor to obtain a shot similar to that depicted below.



[Screenshot_20220508-101827.jpg]<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F37101212%2F167276443-f8a71271-b781-4bd9-8d3a-fb5eb12039fb.jpg&data=05%7C01%7C%7Caf336a76c0344afb31f608da30895bd2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637875663755677254%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=UWQ94l3Oq6GekcJuTqvUSKrngQmBWWg1v8p4j2Nws0s%3D&reserved=0>

Based on this direction, only one camera scene of the platform would be required in OBS.

**Comment by sircharlo** [comment]
> Hello Charles, is it possible to import the OBS scenes in JWMMF in the same order as they are configured in OBS? 

A great point. [Done.](https://github.com/sircharlo/jw-meeting-media-fetcher/commit/23c213e0a90cbb222579fb26b48650583e5008bb)

> A solution can also be if all scenes have a keyboard shortcut like ALT+1, ALT+2 etc.. or any other key

This would be doable, but I think I'll wait a bit to see if it becomes an often-requested feature before putting time in coding it.

**Comment by riemsesy** [comment]
This is already enough. Thanks a lot!

Groeten, Richard
________________________________
Van: Charles Olivier Savignac ***@***.***>
Verzonden: Sunday, May 8, 2022 4:33:54 PM
Aan: sircharlo/jw-meeting-media-fetcher ***@***.***>
CC: riemsesy ***@***.***>; Author ***@***.***>
Onderwerp: Re: [sircharlo/jw-meeting-media-fetcher] imported OBS scenes order / keyboard shortcuts (Discussion #293)


Hello Charles, is it possible to import the OBS scenes in JWMMF in the same order as they are configured in OBS?

A great point. Done.<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsircharlo%2Fjw-meeting-media-fetcher%2Fcommit%2F23c213e0a90cbb222579fb26b48650583e5008bb&data=05%7C01%7C%7C4378fd5ab0454cf08b6608da30ffc7f2%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637876172376123480%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=VFPigG67XbkVRcIxrGiwVcDQiSmgWNcN8%2Fyp2d%2F2KOk%3D&reserved=0>

A solution can also be if all scenes have a keyboard shortcut like ALT+1, ALT+2 etc.. or any other key

This would be doable, but I think I'll wait a bit to see if it becomes an often-requested feature before putting time in coding it.

**Comment by sircharlo** [comment, marked as answer]
Update will be released in 15 minutes with OBS scene order fix

**Comment by joaochris** [comment]
@sircharlo @riemsesy Assigning hotkeys to scenes in the JWMMF is a great idea! A brother got in touch and he and 3 other people asked for this feature, besides me :D. For those who have several scenes configured, this is a very useful resource!

**Reply by sircharlo** [reply]
Have you tried [setting the keyboard shortcuts directly](https://obsproject.com/forum/threads/hotkeys-for-scene-scene-switch.19508/) in OBS Studio? I just don't want to overlap work if not necessary. If keyboard shortcuts can already be assigned in OBS Studio, I would recommend going that way. 

Someone would have to test if the shortcuts work even when the OBS Studio window does not have focus or is minimized, though.

**Reply by sircharlo** [reply]
In the [OBS Studio docs](https://obsproject.com/wiki/OBS-Studio-Overview#settings) it explains how to set a scene hotkey under the Hotkeys section, as well.

**Reply by joaochris** [reply]
@sircharlo True, we can go that way. For now I believe that I would only need to make it so that when we change the scene using the shortcut key in OBS, it also changes in the JWMMF scene selection.

**Reply by sircharlo** [reply]
I hadn't thought of that. Good point. What kind of shortcuts are we taking about here? ALT+1,2,3 etc based on the scene number in the list?

**Reply by joaochris** [reply]
Exactly!

**Reply by riemsesy** [reply]
![image](https://user-images.githubusercontent.com/39587825/167866608-d87db75e-34af-4be0-82f8-dc183857f650.png)

**Reply by riemsesy** [reply]
just one question.. doesn't JW MMF revert back to the default cam, or does JW MMF see the active cam via websocket after playing media.

**Reply by sircharlo** [reply]
Yeah exactly, it reverts to the cam chosen in the dropdown list in the media controller. I'll see if i can add shortcuts ALT+(from 1-9) directly in JWMMF for scene switching. They would then have to be removed from OBS, of course.

**Reply by sircharlo** [reply]
Shortcuts added, alt 1-0.

**Reply by joaochris** [reply]
> Shortcuts added, alt 1-0.

Thank you very much! I've already tested it and it works great! At first it wasn't working kkkk but after trying about 5 times it started to work. At first the shortcut keys were not showing.

**Reply by sircharlo** [reply]
make sure you remove any shortcuts configured in obs studio..

**Reply by joaochris** [reply]
> make sure you remove any shortcuts configured in obs studio..

That was it! Thank you! It was also great with him "listening" to the obs!
